pkg/ml/context/checkpoints/checkpoints.go

```diff
@@ -808,13 +817,16 @@ func (h *Handler) takeMean(baseNames []string) error {
 // Save creates a new checkpoint and save the context variables and (optionally) Params.
 //
 // All variables in the context are saved, as well as those previously loaded -- this allows one
-// to load the variables only for a part of the model, update that part and save again with everything.
+// to load the variables only for a part of the model, update that part, and save again with everything.
 //
 // Params is (de-) serialized with package json.
 //
 // If the handler is nil, this is a no-op: so it's safe to simply be called, even if the user hasn't configured a
 // checkpoint.
-func (h *Handler) Save() error {
+//
+// By default, the binary file is compressed.  The option WithCompression overrides the default behavior.  This
+// information is reported in the JSON file.
+func (h *Handler) Save(opts ...HandlerOptions) error {
```

Comment:
Instead of using option objects commonly used in Go, GoMLX uses the "builder pattern" everywhere for optional parameters.

I was going to suggest that the compression format is selected in the Config object (the builder). Maybe a `func  (c *Config) WithCompression(bf BinFormat) *Config`, with the default value set to gzip, and `Handler.Save()` can use that.

I don't expect one needing to switch save formats mid-training ... We can always later create an alternative Save method for that when/if needed.